lib/routes/account.js

```diff
@@ -1743,21 +1780,59 @@ module.exports = (
         // failed right away.
         request.emitMetricsEvent('email.verify_code.clicked')
 
-        db.account(uid)
-          .then(
-            (account) => {
-              // This endpoint is not authenticated, so we need to look up
-              // the target email address before we can check it with customs.
-              return customs.check(request, account.email, 'recoveryEmailVerifyCode')
-                .then(
-                  () => account
-                )
-            }
-          )
-          .then(
-            (account) => {
+        return db.account(uid)
+          .then((account) => {
+            // This endpoint is not authenticated, so we need to look up
+            // the target email address before we can check it with customs.
+            return customs.check(request, account.email, 'recoveryEmailVerifyCode')
+              .then(() => {
+                return account
+              })
+          })
+          .then((account) => {
+            // Check if param `type` is specified and equal to `secondary`
+            // If so, verify the secondary email and respond
+            if (type && type === 'secondary' && features.isSecondaryEmailEnabled()) {
```

Comment:
@vladikoff Updated this part based on our discussion yesterday. The only caveat is that I still ended up having to query for a user emails. The main reasons was to ensure that the code belonged to the user and to get the email associated with the code so that it could be sent in the post verify email.